### PR TITLE
fix(challenges): put finishing touches on react & redux challenges

### PR DESCRIPTION
**Pull request by no-stack-dub-sack**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.org/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [ ] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [ ] Addressed currently open issue (replace XXXXX with an issue no in next line)

Closes #XXXXX

#### Description

@BerkeleyTrue @QuincyLarson @bonham000 This PR puts some finishing touches on the React and Redux challenges. There were a few formatting issues since @BerkeleyTrue's recent changes which affected some syntax we used, and more importantly, adds back `<div id='challenge-node'></div>` to the React challenge template for the ReactDOM challenges which were broken with the change. 

@BerkeleyTrue Also, and I'm a bit befuddled by this, but apparently the `code` variable now seems to contain the untranspiled ES6 code? This was clearly not the case a few months ago, but @Ethan-Arrowood's recent work on the ES6 challenges highlighted this for me, because he seemed to be able to get the original code strings without using your `getUserInput =>` syntax... so not quite sure when this happened, but since this seems to work, I replaced all of that syntax with `code` because its much more readable and intuitive. 

@QuincyLarson A few other finishing touches and fixes, but everything I've touched on seems to work, but I'm sure the challenges could use a once over QA with someone actually completing each challenge. @Ethan-Arrowood QA'd them for copy and formatting, but I don't believe for working challenges, and since we're not testing them automatically, this would be a good last step if you can find a few volunteers to run through all 3 sections. 

P.S. I couldn't seem to fix this formatting issue with the new async challenge syntax which @BerkeleyTrue just introduced:
 
image

### CODE REVIEW

**Comment by BerkeleyTrue** [comment]
>  I replaced all of that syntax with code because its much more readable and intuitive.

`code` is legacy and should not be used at all as it will eventually be removed. getUserInput should be used instead.

File: seed/challenges/03-front-end-libraries/react-and-redux.json
Comment (BerkeleyTrue) [inline seed/challenges/03-front-end-libraries/react-and-redux.json]: `code` is legacy. We use getUserInput as interface to get the raw user input.

File: seed/challenges/03-front-end-libraries/react.json
Comment (BerkeleyTrue) [inline seed/challenges/03-front-end-libraries/react.json]: This is unnecessary, I've formatted the challenges to use the root div.

File: seed/challenges/03-front-end-libraries/react.json
Comment (BerkeleyTrue) [inline seed/challenges/03-front-end-libraries/react.json]: Why is this removed?

File: seed/challenges/03-front-end-libraries/react.json
Comment (BerkeleyTrue) [inline seed/challenges/03-front-end-libraries/react.json]: Why is this removed?

**Review by BerkeleyTrue — CHANGES_REQUESTED** [review]

File: seed/challenges/03-front-end-libraries/react.json
Comment (no-stack-dub-sack) [reply seed/challenges/03-front-end-libraries/react.json]: @BerkeleyTrue This is not for rendering the component's to the DOM, though it does double as this in this case. It is for the ReactDOM challenges, where campers need to actually include `ReactDOM.render()` in their code. We ask them to render to a div with id `challenge-node`. Removing the tail for those challenges prevents duplicate rendering. We could just ask them to render to `root` instead, but was hoping this would help avoid confusion and thinking they need to do this every time, since we do it in the background.

Comment (BerkeleyTrue) [reply seed/challenges/03-front-end-libraries/react.json]: I see, then this comment can be ignored.

File: seed/challenges/03-front-end-libraries/react-and-redux.json
Comment (no-stack-dub-sack) [reply seed/challenges/03-front-end-libraries/react-and-redux.json]: Ok, we can revert this then. We should also mention this on #16201

Comment (QuincyLarson) [reply seed/challenges/03-front-end-libraries/react-and-redux.json]: @no-stack-dub-sack did you get a chance to revert this? Is there anything else necessary before we can merge this?

Comment (no-stack-dub-sack) [reply seed/challenges/03-front-end-libraries/react-and-redux.json]: I'm about to push the changes! Was offline all day yesterday or would have gotten to it sooner. I think other than this, a final run through by someone independent, running through all challenges and solving them would be a great last step.

**Comment by no-stack-dub-sack** [comment]
@BerkeleyTrue @QuincyLarson This is all set. I think other than this, a final run through by someone independent, running through all challenges and solving them would be a great last step.

**Comment by QuincyLarson** [comment]
@no-stack-dub-sack Awesome! That's great news! I've got dozens of volunteers who are interested in working through the full React + Redux curriculum on beta over the next few days :) Let's go ahead and merge this and campers QA'ing these challenges can open individual issues for any bugs they find.

@BerkeleyTrue what do you think of this plan? If this works for you, can you dismiss your review and merge?

**Comment by QuincyLarson** [comment]
@no-stack-dub-sack Thanks again, man! Excited to get campers working through these. 

Thanks to @BerkeleyTrue for QA'ing.

**Pull request merged**
